scalameter-core/src/main/scala/org/scalameter/Parameter.scala

```diff
@@ -0,0 +1,86 @@
+    override final def skipWhitespace = false
+
+    private lazy val value: PackratParser[AnyValManifest[_]] = {
+      ("Byte" ^^^ Manifest.Byte) |
+        ("Short" ^^^ Manifest.Short) |
+        ("Char" ^^^ Manifest.Char) |
+        ("Int" ^^^ Manifest.Int) |
+        ("Long" ^^^ Manifest.Long) |
+        ("Float" ^^^ Manifest.Float) |
+        ("Double" ^^^ Manifest.Double) |
+        ("Boolean" ^^^ Manifest.Boolean) |
+        ("Unit" ^^^ Manifest.Unit)
+    }
+
+    private lazy val phantom: PackratParser[Manifest[_]] = {
+      ("Any" ^^^ Manifest.Any) |
+        ("Object" ^^^ Manifest.Object) |
+        ("AnyRef" ^^^ Manifest.AnyRef) |
+        ("AnyVal" ^^^ Manifest.AnyVal) |
+        ("Null" ^^^ Manifest.Null) |
+        ("Nothing" ^^^ Manifest.Nothing)
+    }
+
+    private def cls: PackratParser[String] = rep1sep(ident, ".") ^^ { case s =>
+      s.mkString(".")
+    }
+
+    private lazy val singleton: PackratParser[Manifest[_]] = cls ^? { case s if s.endsWith(".type") =>
+      val clazz = Class.forName(s.stripSuffix(".type")+"$")
+      Manifest.singleType(clazz.getField("MODULE$").get(null))
+    }
+
+    private lazy val classType: PackratParser[Manifest[_]] = opt(manifest <~ "#") ~ cls ~ opt("[" ~> rep1sep(manifest, ", ") <~ "]") ^^ {
+      case None ~ s ~ None => Manifest.classType(Class.forName(s))
+      case None ~ s ~ Some(targ :: targs) => Manifest.classType(Class.forName(s), targ, targs: _*)
+      case Some(p) ~ s ~ targs => Manifest.classType(p, Class.forName(s), targs.getOrElse(Nil): _*)
+    }
+
+    private lazy val array: PackratParser[Manifest[_]] = "Array" ~ "[" ~> manifest <~ "]" ^^ { case m =>
+      Manifest.arrayType(m)
+    }
+
+    private lazy val wildcard: PackratParser[Manifest[_]] = "_" ~> (opt(" >: " ~> manifest) ~ opt(" <: " ~> manifest)) ^^ {
+      case None ~ None => Manifest.wildcardType(Manifest.Nothing, Manifest.Any)
+      case None ~ Some(m) => Manifest.wildcardType(Manifest.Nothing, m)
+      case Some(m) ~ None => Manifest.wildcardType(m, Manifest.Any)
+      case Some(m1) ~ Some(m2) => Manifest.wildcardType(m1, m2)
+    }
+
+    private lazy val intersectionType: PackratParser[Manifest[_]] = rep1sep(manifest, " with ") ^^ { case ms =>
+      Manifest.intersectionType(ms: _*)
+    }
+
+    private[Parameter] lazy val manifest: PackratParser[Manifest[_]] =
+      value | phantom | singleton | array | wildcard | classType | intersectionType
+  }
+
+  def fromString(str: String): Parameter[_] = {
+    val i = str.lastIndexOf('|')
+    if (i == -1) sys.error("""Invalid parameter string. It should have following form "name|manifest".""")
+    val key = str.substring(0, i)
+    val manifestString = str.substring(i + 1)
+    val parser = new ManifestParser
+    val m = parser.parse(parser.manifest, new parser.PackratReader(new CharSequenceReader(manifestString))).getOrElse(
+      sys.error(s"""Invalid manifest string: "$manifestString".""")
+    )
+    Parameter(key)(m)
+  }
```

Comment:
I took a look at these parsers, and they seem to do the right thing.

I appreciate the effort you invested into this design, however, I'm not sure if we want to go in this direction. I was really hoping that we could have some serializer that just works out-of-the-box. Here, it looks like we're trying to reimplement the wheel by creating a custom encoding of all the Scala types.
The costs are:
- increased complexity
- increased space consumption
- increased maintenance costs

At the same time, the benefits of supporting any type whatsoever in the Parameters is questionable - the provided basic `Gen` objects mostly create primitive values. Those basic `Gen` objects can compose into more complex data types, but they always need to serialize only the values from the basic `Gen`s.

There are (at most) 1% of users that write custom `Gen` implementations. Of those, hardly anybody uses types other than the primitives and `String`. It seems like it is not worth to add such a complexity only to appease such users - if they insist on custom generators, these users can still use the `SerializationPersistor`.

Here's what I would suggest instead:
- introduce (to the `core` project) a (type-)class `Pickler[T]`:

        abstract class Pickler[T] {
          def pickle(x: T): Array[Byte] // byte array is general enough - serializers can use base64 to escape the world of bytes
          def unpickle(a: Array[Byte]): T
        }

- implement `Pickler`s for the basic data types - primitives and `String`
- ensure that `Gen` factories (especially, `single` and `enumerate`) take the `Pickler[T]` typeclass
- enrich `Parameters` with the `Pickler` arguments - you could replace `Any` in `axisData` with just `(Any, Pickler[Any])` - it is fine if we're internally not always completely type-safe

In the main project:
- introduce your JSON serializer that will encode the `Pickler` classname along with the array contents

We can do something similar if we need to persist `Key`s, but first focus on the `Parameters`.

This way:
- people get a binary-compatible serializer that works with basic data-types
- people who really want more can do their own picklers - we don't need to support exotic picklers for them - or they can revert to `SerializationPersistor`